source/_posts/2015-11-16-immutable-strings-in-ruby-2-dot-3.markdown

```diff
@@ -0,0 +1,136 @@
+---
+layout: post
+title: "Immutable strings in Ruby 2.3"
+category:
+date: 2015-11-16 08:39:25 -0300
+comments: true
+author:
+  name: Alexis Mas
+  email: alexis@wyeworks.com
+  twitter_handle: _axxiss
+  github_handle: axxiss
+  image: /images/team/
+  description: Software Engineer. Turning coffee into software since 2008.
+---
+
+String literals immutability is one of the new features you can find in [Ruby 2.3][1].
+But what does this implies?
+
+* Equality comparison affected
+* Better performance
+* Thread safety
+
+
+Currently strings immutability is included as opt-in, as the first step of a multi
+stage rollout and if it works well it will be [enabled by default on Ruby 3.0][2].
+So it still can end up being just an experiment. Time will tell.
+
+<!--more-->
+
+# Immutability
+
+Quoting [wikipedia][3]:
+
+> An immutable object is an object whose state cannot be modified after it is created.
+
+As immutable objects state cannot be changed it doesn't make sense to copy the
+same object over and over. Following that approach it will have three major
+consequences:
```

Comment:
I think this phrase is a bit confusing. I guess you're trying to explain that when an object does not change along the whole execution of a program it does not make sense to make new copies, right?.

Because it does make sense to copy immutable objects if you need to return a new value derived from them.